client/me/account-close/confirm-dialog.jsx

```diff
@@ -6,68 +6,89 @@
 import React from 'react';
 import { localize } from 'i18n-calypso';
 import { noop } from 'lodash';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import Dialog from 'components/dialog';
 import FormLabel from 'components/forms/form-label';
 import Button from 'components/button';
+import { getCurrentUser } from 'state/current-user/selectors';
 
-const AccountCloseConfirmDialog = ( {
-	isVisible,
-	username,
-	onConfirm,
-	translate,
-	closeConfirmDialog,
-	deleteAccount,
-} ) => {
-	const deleteButtons = [
-		<Button onClick={ closeConfirmDialog }>{ translate( 'Cancel' ) }</Button>,
-		<Button primary scary disabled onClick={ deleteAccount }>
-			{ translate( 'Close your account' ) }
-		</Button>,
-	];
+class AccountCloseConfirmDialog extends React.Component {
+	state = {
+		inputValue: '',
+	};
 
-	return (
-		<Dialog
-			isVisible={ isVisible }
-			buttons={ deleteButtons }
-			className="account-close__confirm-dialog"
-		>
-			<h1 className="account-close__confirm-dialog-header">
-				{ translate( 'Confirm account closure' ) }
-			</h1>
-			<FormLabel htmlFor="confirmAccountCloseInput" className="account-close__confirm-dialog-label">
-				{ translate(
-					'Please type {{warn}}%(username)s{{/warn}} in the field below to confirm. ' +
-						'Your account will then be gone forever.',
-					{
-						components: {
-							warn: <span className="account-close__confirm-dialog-target-username" />,
-						},
-						args: {
-							username: username,
-						},
-					}
-				) }
-			</FormLabel>
+	handleCancel = () => {
+		this.setState( { inputValue: '' } );
+		this.props.closeConfirmDialog();
+	};
 
-			<input
-				autoCapitalize="off"
-				className="account-close__confirm-dialog-confirm-input"
-				type="text"
-				onChange={ onConfirm }
-				aria-required="true"
-				id="confirmAccountCloseInput"
-			/>
-		</Dialog>
-	);
-};
+	handleInputChange = event => {
+		this.setState( { inputValue: event.target.value.toLowerCase() } );
+	};
+
+	render() {
+		const { isVisible, currentUsername, onConfirm, translate } = this.props;
+		const isButtonDisabled = currentUsername && this.state.inputValue !== currentUsername;
+		const deleteButtons = [
+			<Button onClick={ this.handleCancel }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary scary disabled={ isButtonDisabled } onClick={ onConfirm }>
+				{ translate( 'Close your account' ) }
+			</Button>,
+		];
+
+		return (
+			<Dialog
+				isVisible={ isVisible }
+				buttons={ deleteButtons }
+				className="account-close__confirm-dialog"
+			>
+				<h1 className="account-close__confirm-dialog-header">
+					{ translate( 'Confirm account closure' ) }
+				</h1>
+				<FormLabel
+					htmlFor="confirmAccountCloseInput"
+					className="account-close__confirm-dialog-label"
+				>
+					{ translate(
+						'Please type {{warn}}%(currentUsername)s{{/warn}} in the field below to confirm. ' +
```

Comment:
Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Please type {{warn}}%(username)s{{/warn}} in the field below to confirm. Your account will then be gone forever.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).